java/src/jmri/jmrit/beantable/AbstractTableTabAction.java

```diff
@@ -247,6 +247,13 @@ private void fireColumnsUpdated(){
             if (model instanceof XTableColumnModel) {
                 Enumeration<TableColumn> e = ((XTableColumnModel) model).getColumns(false);
                 int numCols = ((XTableColumnModel) model).getColumnCount(false);
+                // XTableColumnModel has been spotted to return a fleeting incorrect
+                // column count, possibly when manager is changed at startup
+                // so we do a sanity check to make sure the models are in synch.
+                if (numCols != dataModel.getColumnCount()){
+                    log.debug("Xtable cols: {} Model cols: {}",numCols,dataModel.getColumnCount());
```

Comment:
If this is an error, I think it would be good to have a clearer message that it's actually an error. For example:
`Error: Incorrext number of columns. Xtable cols: {} Model cols: {}`

And maybe have this as `log.error()` instead of `log.debug()`. What is the severity of this error? If the user doesn't notice anything when this happens, `log.debug()` is probably fine. But if users have problems with it, `log.error()` might be better.

Reply:
There are some recommendations on what each logging level should be used for here:  https://www.jmri.org/help/en/html/doc/Technical/Logging.shtml

Reply:
Thanks both, unsure if an error however nothing we can do anything constructive with so removed, can always re-add if there's an issue.

Reply:
I wasn't against logging. Logging messages is good, so I recommend that you put the logging statement back. My concern was the level of logging and the logging message.

The reason to why I prefer a more clear logging message is that I have had numerous cases in multiple of software where I have got an error message I don't understand and I put that error message into google search engine and I mostly get a reasonable answer that helps me out. It's also much easier to find the error message in the source code if there is some more characters to search for, once a user asks what this message means.

Besides that, this PR looks good. But I'm currently unable to restart Travis and I'm waiting for Bob or Matt to resolve that. It's possible to restart Travis by closing/reopening the PR, or by adding a simple commit (like changing a space), but that's last resort.

Reply:
Thanks, have re-added as debug.
If CI is still struggling to be restarted later, I may add more unit tests to this PR, would be good to get an Abstract level click on every column header to check the sort doesn't error.

Reply:
The problem with Travis seems to have been resolved and I have merged this PR.
But it's great if you add more unit tests!